File: Agencias_Viajes_C#/Forms/FrmLogin.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.SqlClient;
using System.Drawing;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Agencias_Viajes_C_
{
    public partial class FrmLogin : Form
    {
        public FrmLogin()
        {
            InitializeComponent();
        }

        string server = "Data Source = LAPTOP-HAO3D8QC\\SQLEXPRESS; Initial Catalog= BD_Agencia_Viajes; Integrated Security = True ";
        SqlConnection conectar = new SqlConnection();
        string usuario;
        private void btnAgregarUs_Click(object sender, EventArgs e)
        {
            FrmAgregarUsuario agregarUsuario = new FrmAgregarUsuario();
            this.Hide();
            agregarUsuario.Show();
        }
        public static string HashPassword(string password)
        {
            using (SHA256 sha256Hash = SHA256.Create())
            {
                // Convertir la contraseña en un arreglo de bytes
                byte[] bytes = sha256Hash.ComputeHash(Encoding.UTF8.GetBytes(password));

                // Convertir el arreglo de bytes a una cadena hexadecimal
                StringBuilder builder = new StringBuilder();
                for (int i = 0; i < bytes.Length; i++)
                {
                    builder.Append(bytes[i].ToString("x2"));
                }

                return builder.ToString();
            }
        }

        private void btnEntrar_Click(object sender, EventArgs e)
        {
            string password = txtContrasenia.Text;
            conectar.ConnectionString = server;
            conectar.Open();
            string query = "select Contrasena from USUARIO where Nombre_Usuario=@usuario";
            SqlCommand cmd = new SqlCommand(query, conectar);
            cmd.Parameters.AddWithValue("@usuario", txtUsuario.Text);
            object resultado = cmd.ExecuteScalar();
            string hashedpassword = HashPassword(password);
            bool soniguales = string.Equals(resultado, hashedpassword);
            if (soniguales)
            {
                usuario = txtUsuario.Text;
                MessageBox.Show("Login exitoso");
                FrmAgencia formag = new FrmAgencia(usuario);
                formag.Show();
                this.Hide();
                conectar.Close();
            }
            else
            {
                MessageBox.Show("Usuario o Contraseña incorrectos");
            }
            conectar.Close();
        }

        private void FrmLogin_Load(object sender, EventArgs e)
        {

        }
    }
}

File: Agencia_Viajes_VB/Agencia_Viajes_VB/FrmLogin.vb
Imports System.Data.SqlClient
Imports System.Security.Cryptography
Imports System.Text
Imports MySql.Data.MySqlClient

Public Class FrmLogin
    Dim server As String = "Server=localhost; user=root; password=soporte; Database=bd_agencia_viajes;"
    Dim conectar As MySqlConnection = New MySqlConnection(server)
    Dim usuario As String

    Private Sub btnAgregarUs_Click(sender As Object, e As EventArgs) Handles btnAgregarUs.Click
        Dim agregarUsuario As New FrmAgregarUsuario()
        agregarUsuario.Show()
        'Me.Close()
    End Sub

    Public Function HashPassword(ByVal password As String) As String
        Using sha256Hash As SHA256 = SHA256.Create()
            Dim bytes As Byte() = sha256Hash.ComputeHash(Encoding.UTF8.GetBytes(password))
            Dim builder As New StringBuilder()

            For i As Integer = 0 To bytes.Length - 1
                builder.Append(bytes(i).ToString("x2"))
            Next

            Return builder.ToString()
        End Using
    End Function

    Private Sub btnEntrar_Click(sender As Object, e As EventArgs) Handles btnEntrar.Click
        Dim password As String = txtContrasenia.Text


        conectar.ConnectionString = server
        conectar.Open()
        Dim query As String = "SELECT Contrasena FROM USUARIO WHERE Nombre_Usuario=@usuario;"
        Dim cmd As New MySqlCommand(query, conectar)
        cmd.Parameters.AddWithValue("@usuario", txtUsuario.Text)
        Dim resultado As Object = cmd.ExecuteScalar()
        Dim hashedpassword As String = HashPassword(password)
        Dim sonIguales As Boolean = String.Equals(resultado, hashedpassword)

        If sonIguales Then
            usuario = txtUsuario.Text
            MessageBox.Show("Login exitoso")
            Dim formAg As New FrmAgencia(usuario)
            formAg.Show()
            Me.Hide()
            conectar.Close()
        Else
            MessageBox.Show("Usuario o Contraseña incorrectos")
        End If
        
        conectar.Close()


    End Sub


End Class
